spec/helpers/search_helper_spec.rb

```diff
@@ -1,6 +1,68 @@
 require 'rails_helper'
 
 RSpec.describe SearchHelper, type: :helper do
+
+  describe "#paginate" do
+
+    let (:page_params) {{
+      :total_pages=>3,
+      :previous_link=>"/petitions?page=1&state=all",
+      :next_link=>"/petitions?state=all",
+      "model" => Petition
+    }}
+
+    let (:first_page_params) {{
+      "first_page?" => true,
+      "last_page?" => false,
+      "previous_params" => {:state=>:all, :page=>nil},
+      "next_params" => {:state=>:all, :page=>2},
+      :previous_page=>nil,
+      :next_page=>2,
+    }}
+
+    let (:intermediary_page_params) {{
+      "first_page?" => false,
+      "last_page?" => false,
+      "previous_params" => {:state=>:all, :page=>1},
+      "next_params" => {:state=>:all, :page=>3},
+      :previous_page=>1,
+      :next_page=>2,
+    }}
+
+    let (:last_page_params) {{
+      "first_page?" => false,
+      "last_page?" => true,
+      "previous_params" => {:state=>:all, :page=>2},
+      "next_params" => {:state=>:all, :page=>nil},
+      :previous_page=>2,
+      :next_page=>nil,
+    }}
+
+    context "when it handles the first page of petitions" do
+      it "only adds a link to the next page" do
+        petitions = double('petitions', page_params.merge(first_page_params))
+        expect(paginate(petitions)).to include "Next"
+        expect(paginate(petitions)).not_to include "Previous"
+      end
+    end
+
+    context "when it handles an intermediary page of petitions" do
+      it "adds both pagination links" do
+        petitions = double('petitions', page_params.merge(intermediary_page_params))
```

Comment:
Move the petitions to a `let` block inside the `context` and then split the expectations into two parts - "adds a link to the previous page" and "adds a link to the next page". The idiomatic form for RSpec is a single expectation per example though obviously that's flexible where needs must.